### PR TITLE
feat: add FSDP2 multi-GPU training backend with multi-tenant LoRA support

### PR DESCRIPTION
**Pull request by xuanrui-L**
Implement a fully-sharded data-parallel (FSDP2) training backend that distributes base models across multiple GPUs while supporting dynamic multi-tenant LoRA adapter management.

Core components:
- FSDPTrainingWorker: per-GPU worker handling model construction, FSDP2 wrapping (MixedPrecisionPolicy bfloat16/float32), and multi-tenant adapter lifecycle (create/switch/remove) with gradient isolation
- FSDPWorkerGroup: Ray-based orchestrator that manages worker actors, distributes data shards, and aggregates results across ranks
- FSDPTrainingBackend: async interface bridging the training controller to the FSDP worker group
- fsdp_utils: version-aware FSDP2 imports (PyTorch 2.4+), ABC compatibility patching for HuggingFace models, device mesh creation

Key design decisions:
- Pre-wrap all decoder-layer linears with a default LoRA adapter before FSDP2 sharding so that subsequent add_adapter calls use PEFT's update_layer (in-place weight addition) instead of _replace_module (which would break FSDP2 parameter tracking)
- LoRA parameters are regular tensors (not DTensors) and require explicit broadcast (init) and all_reduce (gradients) across ranks
- Per-adapter optimizer instances and saved gradient buffers enable decoupled forward/backward/optim_step across tenant switches

Tests:
- test_fsdp_training: basic loss decrease, multi-tenant isolation, checkpoint save/load, gradient isolation across adapters
- test_fsdp_e2e: end-to-end server + backend tests covering dynamic adapter lifecycle, checkpoint resume, rapid switching, memory stability, decoupled operations, cross-rank parameter sync
- test_fsdp_comprehensive: multi-rank adapters on one model, training state preservation across switches, sequential multi-model training, inference deployment via vLLM, GPU memory lifecycle monitoring

All test model paths are resolved from environment variables (TUFT_TEST_MODEL, FSDP_TEST_MODEL_A/B) with no hardcoded paths. Fixtures enforce >= 2 GPU availability and perform clear_ray_state before and after each test.

### CODE REVIEW

File: src/tuft/backends/fsdp_utils.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+# ---------------------------------------------------------------------------
+# DeviceMesh helpers.
+# ---------------------------------------------------------------------------
+def create_device_mesh(world_size: int, fsdp_size: int = -1):
+    """Create a 1-D or 2-D device mesh for FSDP2.
+
+    Args:
+        world_size: total number of ranks.
+        fsdp_size: FSDP group size. If -1 or >= world_size, use pure FSDP
```
Comment (pan-x-c) [inline src/tuft/backends/fsdp_utils.py]: HSDP (`fsdp_size != -1`) is not used. Remove the `fsdp_size` if not necessary.
